terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -126,6 +153,26 @@ def git_declaration_regex(filename)
           (?:(?!^\}).)*
         /mx
       end
+
+      def lockfile_declaration_regex(provider_source)
+        /
+          (?:(?!^\}).)*
+          provider\s*["']#{Regexp.escape(provider_source)}["']\s*\{
+          (?:(?!^\}).)*}
+        /mx
+      end
+
+      def copy_dir_to_temporary_directory
+        SharedHelpers.in_a_temporary_directory do
+          dependency_files.each do |file|
+            # Do not include the .terraform directory
+            next if file.name.include?(".terraform/")
```

Comment:
I would not expect any of the `dependency_files` to be this directory, when does this happen?

Reply:
Dependency files are copied to a temporary directory to provide context for the `terraform providers lock` command. The files are copied over in the `FileUpdater#update_lockfile_declaration` function but I'm not sure when the `dependency_files` array itself is populated.

Reply:
It's what we grab in the `FileFetcher`, I don't think we ever pull in those files so we probably don't need to worry about it?